src/coreclr/src/pal/src/thread/context.cpp

```diff
@@ -1116,6 +1134,31 @@ CONTEXT_GetThreadContextFromThreadState(
             CONTEXT_GetThreadContextFromThreadState((thread_state_flavor_t)pState->fsh.flavor, (thread_state_t)&pState->ufs, lpContext);
         }
         break;
+#elif defined(HOST_ARM64)
+        case ARM_THREAD_STATE64:
+            if (lpContext->ContextFlags & (CONTEXT_CONTROL | CONTEXT_INTEGER | CONTEXT_SEGMENTS) & CONTEXT_AREA_MASK)
+            {
+                arm_thread_state64_t *pState = (arm_thread_state64_t*)threadState;
+                memcpy(&lpContext->X0, &pState->__x[0], 29 * 8);
+                lpContext->Cpsr = pState->__cpsr;
+                lpContext->Fp = arm_thread_state64_get_fp(*pState);
+                lpContext->Sp = arm_thread_state64_get_sp(*pState);
+                lpContext->Lr = arm_thread_state64_get_lr(*pState);
+                lpContext->Pc = arm_thread_state64_get_pc(*pState);
```

Comment:
There are actually two versions of some of these functions.  i.e. `arm_thread_state64_get_lr` & `arm_thread_state64_get_lr_fptr`.  I believe this is to support Arm64 pointer authentication feature.  I haven't figure out which is which and whether the current AppleSilicon has pointer authentication feature support.  It was an 8.x feature.

Reply:
Looking at the macros, the two versions differ only in the return type.  I chose to consistently use the `arm_thread_state64_get_*_fptr` form when it was available,  This form return a function pointer.